rclpy/src/rclpy/_rclpy.c

```diff
@@ -3369,6 +3370,385 @@ rclpy_clock_set_ros_time_override(PyObject * Py_UNUSED(self), PyObject * args)
+    }
+  } else if (variant->double_value) {
+    type_enum_value = 3;
+    value = PyFloat_FromDouble(*(variant->double_value));
+    if (NULL == value) {
+      return NULL;
+    }
+  } else if (variant->string_value) {
+    type_enum_value = 4;
+    value = PyUnicode_FromString(variant->string_value);
+    if (NULL == value) {
+      return NULL;
+    }
+  } else if (variant->byte_array_value) {
+    type_enum_value = 5;
+    value = PyList_New(variant->byte_array_value->size);
+    if (NULL == value) {
+      return NULL;
+    }
+    for (size_t i = 0; i < variant->byte_array_value->size; i++) {
+      member_value = PyBytes_FromFormat("%u", variant->byte_array_value->values[i]);
+      if (NULL == member_value) {
+        Py_DECREF(value);
+        return NULL;
+      }
+      PyList_SET_ITEM(value, i, member_value);
+    }
+  } else if (variant->bool_array_value) {
+    type_enum_value = 6;
+    value = PyList_New(variant->bool_array_value->size);
+    if (NULL == value) {
+      return NULL;
+    }
+    for (size_t i = 0; i < variant->bool_array_value->size; i++) {
+      member_value = variant->bool_array_value->values[i] ? Py_True : Py_False;
+      Py_INCREF(member_value);
+      PyList_SET_ITEM(value, i, member_value);
+    }
+  } else if (variant->integer_array_value) {
+    type_enum_value = 7;
+    value = PyList_New(variant->integer_array_value->size);
+    if (NULL == value) {
+      return NULL;
+    }
+    for (size_t i = 0; i < variant->integer_array_value->size; i++) {
+      member_value = PyLong_FromLongLong(variant->integer_array_value->values[i]);
+      if (NULL == member_value) {
+        Py_DECREF(value);
+        return NULL;
+      }
+      PyList_SET_ITEM(value, i, member_value);
+    }
+  } else if (variant->double_array_value) {
+    type_enum_value = 8;
+    value = PyList_New(variant->double_array_value->size);
+    if (NULL == value) {
+      return NULL;
+    }
+    for (size_t i = 0; i < variant->double_array_value->size; i++) {
+      member_value = PyFloat_FromDouble(variant->double_array_value->values[i]);
+      if (NULL == member_value) {
+        Py_DECREF(value);
+        return NULL;
+      }
+      PyList_SET_ITEM(value, i, member_value);
+    }
+  } else if (variant->string_array_value) {
+    type_enum_value = 9;
+    value = PyList_New(variant->string_array_value->size);
+    if (NULL == value) {
+      return NULL;
+    }
+    for (size_t i = 0; i < variant->string_array_value->size; i++) {
+      member_value = PyUnicode_FromString(variant->string_array_value->data[i]);
+      if (NULL == member_value) {
+        Py_DECREF(value);
+        return NULL;
+      }
+      PyList_SET_ITEM(value, i, member_value);
+    }
+  }
+
+  PyObject * args = Py_BuildValue("(i)", type_enum_value);
+  if (NULL == args) {
+    return NULL;
+  }
+  PyObject * type = PyObject_CallObject(parameter_type_cls, args);
+  Py_DECREF(args);
+  args = Py_BuildValue("OOO", name, type, value);
+  if (NULL == args) {
```

Comment:
should Py_DECREF type and value

Reply:
Now decref'd as of b5fd549bd0a422a9b9aa73a739df1e0c2897f432